YesterdayApi/Core/Email/Builder/Builders/EmailBuilder.cs

```diff
@@ -0,0 +1,19 @@
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using MimeKit;
+using YesterdayApi.Core.Email.Builder.Renderer;
+
+namespace YesterdayApi.Core.Email.Builder.Builders
+{
+    public abstract class EmailBuilder
+    {
+        public EmailBuilder(
+            IRazorViewToStringRenderer razorViewToStringRenderer)
+        {
+        }
+
+        public abstract void ResetEmail();
+        public abstract void AppendUserDetails(IUserCredentials credentials);
+        public abstract void PrepareBody();
+        public abstract MimeMessage BuildEmail();
```

Comment:
They do not provide implementation because they are abstract. This interface wouldn't be used anywhere else and just create new code. The methods wouldn't have a implementation in the abstract class anyways. So what would be the point of putting it to the Interface?

Reply:
Yeah you're right on this one, just wanted to test you xD

Reply:
Actually you @AlexTorianyk thinking is not that bad. Take a look how this abstract class looks like. It doesn't provide ANY implementation. It's empty. For me this should be an interface.

Reply:
Im still using the base constructor, having that, there is no need to create a additional interface.